source/plugin/modularization.rst

```diff
@@ -0,0 +1,61 @@
+===================
+Code Modularization
+===================
+
+When writing Sponge plugins, you may find that you need to modularize your code. Maybe you want to split your core into
+another project so that others can make use of it, or maybe you want to utilize a service another plugin provides. With
+Sponge, begone the days of shading or shadowing entire plugins into your plugin.
+
+Problem 1: ClassNotFoundException
+=================================
+
+Let's say you want to provide some awesome serice for ``WarpService``. You took a day or two to write
+``AwesomeWarpService`` that uses SQL to back warps. So now you can override some other plugin's ``CoreWarpService`` that
+uses mere flatfiles with your own service.
+
+So you go to use ``Sponge.getServiceManager().provide(WarpService.class)``, but you are shocked to find that referencing
```

Comment:
This is where the problem happens, so I think it's appropriate to start this with "But" or "However". Also, please don't mix tenses (this is in future and present tense atm). Keeping to present tense, the last bit should read "warpservice throws a class...".   Do we really need "Oh noes!" ? It may translate badly ;)